subsys/bluetooth/host/conn.c

```diff
@@ -698,6 +698,28 @@ static void conn_cleanup(struct bt_conn *conn)
 	k_work_reschedule(&conn->deferred_work, K_NO_WAIT);
 }
 
+static void conn_destroy(struct bt_conn *conn, void *data)
+{
+	sys_slist_init(&conn->tx_pending);
+	conn->pending_no_cb = 0;
+
+	conn_cleanup(conn);
+
+	bt_conn_set_state(conn, BT_CONN_DISCONNECTED);
+
+	k_work_cancel_delayable(&conn->deferred_work);
```

Comment:
Shouldn't this work be cancelled before other actions performed on `bt_conn`?

Reply:
That makes sense, yes